saleor/discount/utils.py

```diff
@@ -535,6 +721,41 @@ def get_variants_to_promotion_rules_map(
     return rules_info_per_variant
 
 
+def fetch_promotion_rules_for_checkout(
+    checkout: Checkout,
+):
+    from ..graphql.discount.utils import PredicateObjectType, filter_qs_by_predicate
+
+    applicable_rules = []
+    promotions = Promotion.objects.active()
+    checkout_channel_id = checkout.channel_id
+    PromotionRuleChannels = PromotionRule.channels.through.objects.filter(
+        channel_id=checkout_channel_id
+    )
+    rules = (
+        PromotionRule.objects.filter(
+            Exists(promotions.filter(id=OuterRef("promotion_id"))),
+            Exists(PromotionRuleChannels.filter(promotionrule_id=OuterRef("id"))),
+        )
+        .exclude(order_predicate={})
+        .prefetch_related("channels")
```

Comment:
Is this needed? (I just glanced through the function calls, so I might have missed though)

If it is used, then please note, that iterator is compatible with `prefetch_related` if `chunk_size` is set.

Reply:
I removed `prefetch_related` as it's not needed anymore :)